### ISSUE TITLE
Please check social logins, they are not working. (Facebook, twitter)

### ISSUE THREAD

**Issue by otulga** [CakeDC/users#412]
Sign in with Twitter button is working but there is some issue with undefined index 'provider'.

Also Sign in with Facebook button seems not working just redirecting back to login page.

CakePHP: 3.2.13
Users: 3.2.2

**Update 1**
I am very new with github issue so sorry for this type of issue. 

After several hours of debugging I did the following things to work both twitter and facebook login. 
I downgraded Users plugin to 3.1.6
With Users 3.1.6 Facebook login was working. But same with twitter. 

**I did following changes to able to login via twitter.**

_on SocialAuthenticate.php_
`if (!empty($data) && (!empty($data['email']) || !empty($requestDataEmail))) {`
**to**
`if (!empty($data) && (!empty($requestDataEmail))) {`

_on SocialBehavior.php_
`$accountData['avatar'] = str_replace('normal', 'square', $accountData['avatar']);`
**to**
`$accountData['avatar'] = str_replace('normal', '400x400', $accountData['avatar']);`

_I thought twitter made a little change on an avatar link. And i changed the last code to get larger avatar._

I will be very happy for any further advice using Users plugin for cakephp. 
Thank you

**Comment by steinkel** [comment]
Please could you paste the trace for this error "Sign in with Twitter button is working but there is some issue with undefined index 'provider'."

The plugin should work with both providers (twitter, facebook) using the most recent version... Thanks!

**Comment by otulga** [comment]
Sorry for the late reply @steinkel,

```
Notice (8): Undefined index: provider [ROOT/vendor/cakedc/users/src/Model/Behavior/SocialBehavior.php, line 47]
InvalidArgumentException

⟩ CakeDC\Users\Model\Behavior\SocialBehavior->socialLogin
[internal function]
⟩ call_user_func_array
CORE/src/ORM/BehaviorRegistry.php, line 237
⟩ Cake\ORM\BehaviorRegistry->call
CORE/src/ORM/Table.php, line 2001
⟩ Cake\ORM\Table->__call
ROOT/vendor/cakedc/users/src/Auth/SocialAuthenticate.php, line 434
⟩ App\Model\Table\AppUsersTable->socialLogin
ROOT/vendor/cakedc/users/src/Auth/SocialAuthenticate.php, line 434
⟩ CakeDC\Users\Auth\SocialAuthenticate->_socialLogin
ROOT/vendor/cakedc/users/src/Auth/SocialAuthenticate.php, line 307
⟩ CakeDC\Users\Auth\SocialAuthenticate->_touch
ROOT/vendor/cakedc/users/src/Auth/SocialAuthenticate.php, line 373
⟩ CakeDC\Users\Auth\SocialAuthenticate->getUser
ROOT/vendor/cakedc/users/src/Auth/SocialAuthenticate.php, line 157
⟩ CakeDC\Users\Auth\SocialAuthenticate->authenticate
CORE/src/Controller/Component/AuthComponent.php, line 786
⟩ Cake\Controller\Component\AuthComponent->identify
ROOT/vendor/cakedc/users/src/Controller/Traits/LoginTrait.php, line 57
⟩ CakeDC\Users\Controller\UsersController->twitterLogin
[internal function]
⟩ call_user_func_array
CORE/src/Controller/Controller.php, line 435
⟩ Cake\Controller\Controller->invokeAction
CORE/src/Routing/Dispatcher.php, line 115
⟩ Cake\Routing\Dispatcher->_invoke
CORE/src/Routing/Dispatcher.php, line 88
⟩ Cake\Routing\Dispatcher->dispatch
ROOT/webroot/index.php, line 37

```

**Comment by ajibarra** [comment]
Closed by #418
